RealmSwift/Tests/RealmCollectionTypeTests.swift

```diff
@@ -689,7 +689,7 @@ class ResultsWithCustomInitializerTest: TestCase {
         let expected = Array(collection.map { $0.stringCol })
         let actual = collection.value(forKey: "stringCol") as! [String]!
         XCTAssertEqual(expected, actual!)
-        XCTAssertEqual(collection.map { $0 }, collection.value(forKey: "self") as! [CTTStringObjectWithLink])
+        assertEqualObjectCollections(collection.map { $0 }, collection.value(forKey: "self") as! [SwiftCustomInitializerObject])
```

Comment:
May as well take the opportunity to rename it now.

Not sure why that test didn't fail before. Probably something to do with how XCTAssertEqual is implemented? The more I look at the old version of this test case the more confused I am.